src/lib/scanner.ts

```diff
@@ -39,53 +50,46 @@ export class ScanRunner {
      */
     public async runDfa(targets: string[], projectDir: string): Promise<string> {
         // Create the arg array.
-        const args: string[] = this.createDfaArgArray(targets, projectDir);
+        const args: Inputs = this.createDfaArgArray(targets, projectDir);
 
         // Invoke the scanner.
-        const executionResult: ExecutionResult = await this.invokeAnalyzer(args);
+        const executionResult: AnyJson = await this.invokeAnalyzerWithoutCli(args);
 
         // Process the results.
-        return this.processDfaResults(executionResult);
+        return executionResult as string;
```

Comment:
Is the AnyJson returned always a string? If not then this cast will fail in non-string cases. We'll need to double check.

Reply:
Yes, AnyJson is always returned a string for DFA case. That's because we request the output to be in HTML format (as opposed to JSON for the non-DFA case). It'll be good to revisit the output type while designing the core package to accommodate this. (Leaving this comment open for visibility).